scipy/special/_legacy.pxd

```diff
@@ -50,23 +58,23 @@ cdef inline double ellip_harmonic_unsafe(double h2, double k2, double n,
     _legacy_cast_check("_ellip_harm", n, p)
     return ellip_harmonic(h2, k2, <int>n, <int>p, l, signm, signn)
 
-cdef inline double bdtrc_unsafe(double k, double n, double p) nogil:
-    if npy_isnan(k) or npy_isnan(n):
+cdef inline double bdtr_unsafe(double k, double n, double p) nogil:
+    if npy_isnan(k) or npy_isnan(n) or npy_isinf(n):
```

Comment:
The warning should occur before this line-you should always get a warning for hitting this signature.

Reply:
I'm assuming what you mean here is to remove the `if npy_isnan(k) or npy_isnan(n) or npy_isinf(n):` line entirely and adjust indent etc. 
when I do that the `nan`s pass through and b/c they are floats they trigger the deprecation which causes the tests in `scipy/special/tests/test_nan_inputs.py` to fail. 
Maybe I am not understanding what you are suggesting?

Reply:
> which causes the tests in scipy/special/tests/test_nan_inputs.py to fail.

Ah yeah, that. I'd add an extra filter for deprecation warnings in that test, similar to what is done for `RuntimeWarning`. When I wrote those tests this situation hadn't arisen, so I didn't filter for it.

Reply:
Yeah, I'm just seeing your response to these now, I guess I might check ***before*** I do a push. I added an explicit
```
        with warnings.catch_warnings():
            warnings.simplefilter("ignore", DeprecationWarning)
```
in each of the spots where the deprecations are failing tests. Let me know if you have a better way of doing it, I couldn't seem to get the pytest context manager to suppress DeprecationWarnings but the warnings approach worked.